bugsnag-plugin-android-exitinfo/src/main/java/com/bugsnag/android/ExitInfoPluginStore.kt

```diff
@@ -9,39 +10,72 @@ internal class ExitInfoPluginStore(config: ImmutableConfig) {
     private val file: File = File(config.persistenceDirectory.value, "bugsnag-exit-reasons")
     private val logger: Logger = config.logger
     private val lock = ReentrantReadWriteLock()
+    private val isFirstRun: Boolean = !file.exists()
 
-    fun persist(pid: Int) {
+    fun persist(currentPid: Int, exitInfoKeys: Set<ExitInfoKey>) {
         lock.writeLock().withLock {
             try {
-                val text = pid.toString()
-                file.writeText(text)
+                file.writer().buffered().use { writer ->
+                    JsonStream(writer).use { json ->
+                        json.beginObject()
+                            .name("pid").value(currentPid)
+                            .name("exitInfoKeys")
+                        json.value(exitInfoKeys)
+                        json.endObject()
+                    }
+                }
             } catch (exc: Throwable) {
                 logger.w("Unexpectedly failed to persist PID.", exc)
             }
         }
     }
 
-    fun load(): Int? {
-        return lock.readLock().withLock {
-            try {
-                loadImpl()
-            } catch (exc: Throwable) {
-                logger.w("Unexpectedly failed to load PID.", exc)
-                null
-            }
+    fun load(): Pair<Int?, Set<ExitInfoKey>> {
+        if (isFirstRun) {
+            return null to emptySet()
         }
+        return tryLoadJson() ?: (tryLoadLegacy() to emptySet())
     }
 
-    private fun loadImpl(): Int? {
-        if (!file.exists()) {
+    private fun tryLoadJson(): Pair<Int?, Set<ExitInfoKey>>? {
+        try {
+            val fileContents = file.readText()
+            val jsonObject = JSONObject(fileContents)
+            val currentPid = jsonObject.getInt("pid")
+            val exitInfoKeys = mutableSetOf<ExitInfoKey>()
+            val jsonArray = jsonObject.getJSONArray("exitInfoKeys")
+            for (i in 0 until jsonArray.length()) {
+                val exitInfoKeyObject = jsonArray.getJSONObject(i)
+                exitInfoKeys.add(
+                    ExitInfoKey(
+                        exitInfoKeyObject.getString("pid").toInt(),
+                        exitInfoKeyObject.getString("timestamp").toLong()
+                    )
+                )
+            }
+            return Pair(currentPid, exitInfoKeys)
+        } catch (exc: Throwable) {
+            logger.w("Unexpectedly failed to load PID.", exc)
```

Comment:
the `load` function will fallback to the `tryLoadLegacy` function, so this logging is just noise